aggsender/db/aggsender_db_storage.go

```diff
@@ -521,13 +522,25 @@ func (a *AggSenderSQLStorage) SaveNonAcceptedCertificate(
 	shouldRollback := true
 	defer func() {
 		if shouldRollback {
+			a.logger.Infof("saveNonAcceptedCertificate Rolling back transaction")
 			if errRllbck := tx.Rollback(); errRllbck != nil {
 				a.logger.Errorf(errWhileRollbackFormat, errRllbck)
 			}
 		}
 	}()
-
-	raw, err := json.Marshal(nonAcceptedCert)
+	filename := nonAcceptedCertFilename
+	fullPathFilename, err := a.saveSignedCertificateToFile(filename, nonAcceptedCert.SignedCertificate)
+	if err != nil {
+		return fmt.Errorf("saveNonAcceptedCertificate: failed to save signed certificate to file: %w", err)
+	}
+	entry := &NonAcceptedCertificate{
+		Height:            nonAcceptedCert.Height,
+		CreatedAt:         nonAcceptedCert.CreatedAt,
+		Error:             nonAcceptedCert.Error,
+		SignedCertificate: "@" + fullPathFilename,
+		CertificateHash:   common.BytesToHash(keccak256.Hash([]byte(nonAcceptedCert.SignedCertificate))),
```

Comment:
Simpler to reuse the `Keccak256` function from the go ethereum, since we are relying on its data structure for hash and then no need to convert anything from bytes to hash etc.
```suggestion
		CertificateHash:   crypto.Keccak256Hash([]byte(nonAcceptedCert.SignedCertificate)),
```

Reply:
As a side note, it seems that we could get rid off the `github.com/iden3/go-iden3-crypto` dependency, since the `keccak256` calculation can be reused from go ethereum's crypto package, and the `merkletree` package where it is also used, seem to not be in use in the aggkit, so maybe take this opportunity and remove the dependency and the `merkletree` package altogether (it can be a separate PR for sure). WDYT?

Reply:
This library is used for serval syncer for calculate `keccak256`, but MT use `poseidon`